src/genesis/observers/balances.py

```diff
@@ -64,13 +68,60 @@ def __init__(self, db_conn: Connection, on_completed=None, on_error=None) -> Non
                                                 on_completed=on_completed,
                                                 on_error=on_error)
 
+    def _get_name_and_index(self, e: UniqueViolation, balances: List[Balance]) -> Tuple[str, Tuple[int, int]]:
+        # Extract account name and coin from error string
+        duplicate_balance = str(e).split("(")[2].split(")")[0]
```

Comment:
I think it would improve readability just a bit to add `_id` like you had done in the accounts side.

```suggestion
        duplicate_balance_id = str(e).split("(")[2].split(")")[0]
```